streams/src/main/java/org/apache/kafka/streams/state/internals/ChangeLoggingKeyValueBytesStore.java

```diff
@@ -76,6 +87,10 @@ public long approximateNumEntries() {
     public void put(final Bytes key,
                     final byte[] value) {
         wrapped().put(key, value);
+        if (context != null && context.recordMetadata().isPresent()) {
```

Comment:
@vvcephei when can the context be null? I copied this check from your POC but it wasn't everywhere so I am not sure where we need it. I guess the safest is to add it everywhere